versioned/storage/store/src/main/java/org/projectnessie/versioned/storage/versionstore/BaseCommitHelper.java

```diff
@@ -647,4 +649,13 @@ MergeResult<Commit> mergeTransplantSuccess(
 
     return mergeResult.resultantTargetHash(objIdToHash(newHead)).wasApplied(true).build();
   }
+
+  Commit commitObjToCommit(CommitObj newCommit) throws ReferenceNotFoundException {
+    try {
+      ContentMapping contentMapping = new ContentMapping(persist);
+      return contentMapping.commitObjToCommit(true, newCommit);
+    } catch (ObjNotFoundException e) {
+      throw referenceNotFound(e);
```

Comment:
This (or the only call size) could just re-throw it as a `RuntimeException`. Hitting this exception means that something went terribly wrong: content value stored, but cannot be retrieved - that's a bad situation.
Whereas `ReferenceNotFoundExcepiton` tends to indicate a user-error